clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskHealthCachingAgent.java

```diff
@@ -138,72 +141,127 @@ public TaskHealthCachingAgent(
           continue;
         }
 
-        TaskHealth taskHealth;
+        List<TaskHealth> taskHealth = new LinkedList<>();
```

Comment:
Given that the task health cache key is accountName + region + taskId, won't only one of the task healths for this task will be cached?  So you would need to do the "AND" operation across the task's individual load balancer health statuses to calculate a single TaskHealth for this task.

Reply:
Addressed